### PR TITLE
Rejig teacher training adviser flow to allow for primary candidates to also get an adviser instead of stopping the flow

### PR DESCRIPTION
**Pull request by StackedPancakez**
### Trello card
[5118](https://trello.com/c/GPay3ddg/5118-support-expansion-of-rtta-adviser-service-for-returners-to-include-primary)

### Context
Returning teachers who select primary during the flow will no longer be turned away, but will instead now be taken through the rest of the flow to get an adviser.

### Changes proposed in this pull request
Updates to the teacher training adviser wizard and steps as well as introducing a new step, and new help text to some steps.

### Guidance to review
Standard code review (although be aware most is existing code and this is mainly modifying order / conditionals to allow for certain bits to show), and checking the flow. Perhaps one of the most important things to check however is the non-RTTA adviser funnels, in case of any impact / changes on them.

### CODE REVIEW

File: app/views/teacher_training_adviser/steps/_has_teacher_id.html.erb [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
+
+<details class="govuk-details" data-module="govuk-details">
+  <summary class="govuk-details__summary">
+    <span class="govuk-details__summary-text">
+      What is a teacher reference number (TRN)?
+    </span>
+  </summary>
+  <div class="govuk-details__text">
+    A TRN is a 7 digit number that you get if you hold qualified teacher status in England or Wales or you contribute to the Teachers' Pension Scheme.<br>
+    If you're eligible for a TRN in England, you should be allocated one automatically.
```
Comment (tejasjinaraldfe) [inline app/views/teacher_training_adviser/steps/_has_teacher_id.html.erb:14]: Please add another line break here. It doesn't appear consistent.

File: app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
 <% end %>
+
+<details class="govuk-details" data-module="govuk-details">
+  <summary class="govuk-details__summary">
+    <span class="govuk-details__summary-text">
+      How do I find my teacher reference number (TRN)?
+    </span>
+  </summary>
+  <div class="govuk-details__text">
+    Your TRN is sometime on your payslip, pension statement or teacher training records.<br>
```
Comment (tejasjinaraldfe) [inline app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb:14]: Your TRN is **sometimes** on your payslip - **s** is missing

File: app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb [hunk 3]
```diff
@@ -1,5 +1,25 @@
...
+    Alternatively, you can find a lost TRN<br><br>
```
Comment (tejasjinaraldfe) [inline app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb:15]: Please add hyperlink to **find a lost TRN** - https://find-a-lost-trn.education.gov.uk/start

File: app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb [hunk 4]
```diff
@@ -1,5 +1,25 @@
...
+    You'll be asked for your:
+    <ul>
+      <li>name</li>
+      <li>Date of birth</li>
+      <li>National Insurance number</li>
+    </ul>
+    <br><br>
```
Comment (tejasjinaraldfe) [inline app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb:22]: I think these two breaks are adding additional space between the texts and hence it looks a bit inconsistent. Could you please remove and try if it works?

File: app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb [hunk 5]
```diff
@@ -1,5 +1,25 @@
...
+<details class="govuk-details" data-module="govuk-details">
+  <summary class="govuk-details__summary">
+    <span class="govuk-details__summary-text">
+      How do I find my teacher reference number (TRN)?
+    </span>
+  </summary>
+  <div class="govuk-details__text">
+    Your TRN is sometime on your payslip, pension statement or teacher training records.<br>
+    Alternatively, you can find a lost TRN<br><br>
+    You'll be asked for your:
```
Comment (tejasjinaraldfe) [inline app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb:16]: @MaxineCl could you please confirm if having **Alternatively, you can [find a lost TRN](https://find-a-lost-trn.education.gov.uk/start). You'll be asked for your:** in a single line makes sense? If yes, John, could you please make the change?

**Comment by tejasjinaraldfe** [comment]
Erroring when I select Primary as preferred teaching stage. Attaching screenshot for reference.
screencapture-get-into-teaching-app-review-3687-test-teacherservices-cloud-teacher-training-adviser-sign-up-review-answers-2023-11-13-09_44_03

File: app/views/teacher_training_adviser/steps/_has_teacher_id.html.erb [hunk 6]
```diff
@@ -1,6 +1,16 @@
...
 <% end %>
+
+<details class="govuk-details" data-module="govuk-details">
+  <summary class="govuk-details__summary">
+    <span class="govuk-details__summary-text">
+      What is a teacher reference number (TRN)?
+    </span>
+  </summary>
+  <div class="govuk-details__text">
+    A TRN is a 7 digit number that you get if you hold qualified teacher status in England or Wales or you contribute to the Teachers' Pension Scheme.<br><br>
```
Comment (caitlin-reidy) [inline app/views/teacher_training_adviser/steps/_has_teacher_id.html.erb:13]: A TRN is a 7 digit number you automatically get if you:

- hold qualified teacher status (QTS) in England 
- hold early years teacher status (EYTS) in England
- hold a national professional qualification (NPQ)
- are working towards QTS, EYTS or a national professional qualification (NPQ)
- contribute to the Teachers’ Pensions scheme in England 
- have a relevant restriction in relation to teaching in England

File: app/views/teacher_training_adviser/steps/_has_teacher_id.html.erb (hunk 1 above)
Comment (caitlin-reidy) [reply app/views/teacher_training_adviser/steps/_has_teacher_id.html.erb:14]: delete this line

File: app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb [hunk 7]
```diff
@@ -1,5 +1,23 @@
...
 <% end %>
+
+<details class="govuk-details" data-module="govuk-details">
+  <summary class="govuk-details__summary">
+    <span class="govuk-details__summary-text">
+      How do I find my teacher reference number (TRN)?
+    </span>
+  </summary>
+  <div class="govuk-details__text">
+    Your TRN is sometimes on your payslip, pension statement or teacher training records.<br>
```
Comment (caitlin-reidy) [inline app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb:12]: You can find your TRN on:
- a payslip or pension statement
- teacher training records
Or
- You can use the [Find a lost TRN](https://find-a-lost-trn.education.gov.uk/start) service

File: app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb [hunk 8]
```diff
@@ -1,5 +1,23 @@
...
+    Alternatively, you can <%= link_to "find a lost TRN", "https://find-a-lost-trn.education.gov.uk/start" %><br><br>
```
Comment (caitlin-reidy) [inline app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb:13]: delete

File: app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb [hunk 9]
```diff
@@ -1,5 +1,23 @@
...
+    You'll be asked for your:
```
Comment (caitlin-reidy) [inline app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb:14]: delete

File: app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb [hunk 10]
```diff
@@ -1,5 +1,23 @@
...
+    <ul>
+      <li>name</li>
+      <li>Date of birth</li>
+      <li>National Insurance number</li>
+    </ul>
+    <br>
+    If there's a match, you'll get your TRN straight away. If there's no match, or you don't have a National Insurance number, it might take up to 5 days to find your TRN.
```
Comment (caitlin-reidy) [inline app/views/teacher_training_adviser/steps/_previous_teacher_id.html.erb:21]: delete

**Review by caitlin-reidy — CHANGES_REQUESTED** [review]

**Comment by tejasjinaraldfe** [comment]
@StackedPancakez getting a 500 error after I select 'Yes' on 'Do you have a teacher reference number (TRN)?' step to move forward.

**Review by asatwal — DISMISSED** [review]
👍 @martyn-w  - I've done a technical review only and the Ruby code all looks good to me. I haven't looked at or tested the functionality and behaviour of the website. I can see that has been done by others reviewing this PR.

**Review by asatwal — APPROVED** [review]
👍

**Review by tejasjinaraldfe — APPROVED** [review]
Feature is working as expected for RTTA, TTA and ETA candidates.

Content for both 'How do I find my teacher reference number (TRN)?' and 'What is a teacher reference number (TRN)?' is appearing properly.

Approving the PR. cc @martyn-w @gemmadallmandfe @caitlin-reidy

**Pull request merged**